src/qh3/tls.py

```diff
@@ -1803,6 +1841,45 @@ def _client_handle_finished(self, input_buf: Buffer, output_buf: Buffer) -> None
         )
         next_enc_key = self.key_schedule.derive_secret(b"c ap traffic")
 
+        if self._certificate_request is not None:
+            with push_message(self.key_schedule, output_buf):
+                certificate_list = []
+                if self.certificate is not None:
+                    certificate_list.append(self.certificate)
+                certificate_list += self.certificate_chain
```

Comment:
This can be simplified.
That should fail we end up with no certificates to push.

Reply:
AlertHandshakeFailure("No certificate passed for client CertificateVerify"),

Reply:
Maybe I mis-interpreted, but according to the [RFC](https://datatracker.ietf.org/doc/html/rfc8446#section-4.4.2):
```
The client MUST send a Certificate message if and only if the server
   has requested client authentication via a CertificateRequest message
   ([Section 4.3.2](https://datatracker.ietf.org/doc/html/rfc8446#section-4.3.2)).  If the server requests client authentication but no
   suitable certificate is available, the client MUST send a Certificate
   message containing no certificates (i.e., with the "certificate_list"
   field having length 0).  A Finished message MUST be sent regardless
   of whether the Certificate message is empty.
```

Reply:
Good catch! Indeed,

Reply:
Could you simplify those lines, 
It could be written in a comprehension for loop.

Reply:
Would you rather it would be:
```python
[x for x in [self.certificate] + self.certificate_chain if x is not None]
```
I think the possibility of `self.certificate` being `None` making this line less readable.

Reply:
Arguably, 
if `x` is renamed to cert, 
It would be more readable.

I am in favor of the one-line-compr-loop. To be consistent with other places in the code.

Reply:
Though you resolved it, I don't you having any discounts regarding it :)
Refactored it. I used the `x` before actually just because I saw most places use `x`, so I figured this was the project's convention on such enumerations.